### PR TITLE
Start a donation without sign-in (#338)

### PR DESCRIPTION
**Pull request by kumar303**
Supports https://github.com/mozilla/payments-ui/issues/338

### CODE REVIEW

File: public/js/apps/transaction/app.jsx [hunk 1]
```diff
@@ -50,6 +51,10 @@ export default class TransactionApp extends Component {
     var state = this.state;
     var SignIn = this.props.SignIn;
     var Transaction = this.props.Transaction;
+    var product = products.get(state.productId);
+    var signInRequired = product.seller.kind !== 'donations';
+    console.log('sign-in required to transact product?', state.productId,
+                signInRequired ? 'Yes': 'No');
```
Comment (andymckay) [inline public/js/apps/transaction/app.jsx]: this looks cool, i think we'll need to think about making explicit authentication restrictions in payments-config and not infer it from anything to cope with non-authenticated subscriptions. seperate bug though...

**Comment by andymckay** [comment]
looks like a couple of eslint issues

**Comment by andymckay** [comment]
r+with travis fixes

File: public/js/apps/transaction/app.jsx (hunk 1 above)
Comment (kumar303) [inline public/js/apps/transaction/app.jsx]: I think of this still as a strawman type of patch. A few things we can do later:

- make sign-in optional for donations anyway
- maybe rely on the API to see if we need to sign-in: try to make a request and see if we get a 403 before we try to sign in

**Pull request merged**